bc_obps/registration/tests/endpoints/v1/_operations/_operation_id/test_facilities.py

```diff
@@ -113,3 +114,95 @@ def test_facilities_endpoint_list_facilities_with_filter(self):
         assert response.json().get('count') == 1
         assert response_items_3[0].get('name') == name_to_filter
         assert response_items_3[0].get('type') == Facility.Types.LARGE_LFO
+
+    # AUTHORIZATION
+
+    def test_unauthorized_roles_cannot_create_new_facility(self):
+        mock_facility = {
+            'name': 'zip',
+            'type': 'Large Facility',
+            'latitude_of_largest_emissions': 5,
+            'longitude_of_largest_emissions': 5,
+            'operation_id': operation_baker().id,
+        }
+        # IRC users can't post
+        for role in ['cas_pending', 'cas_admin', 'cas_analyst']:
+            response = TestUtils.mock_post_with_auth_role(
+                self, role, self.content_type, mock_facility, custom_reverse_lazy("create_facility")
+            )
+            assert response.status_code == 401
```

Comment:
Do we also need to check this for unapproved industry users?

Reply:
Added a test